hare3/hare.go

```diff
@@ -195,13 +196,19 @@ type Hare struct {
 	pubsub    pubsub.PublishSubsciber
 	db        *datastore.CachedDB
 	verifier  *signing.EdVerifier
-	signer    *signing.EdSigner
 	oracle    *legacyOracle
 	sync      system.SyncStateProvider
 	patrol    *layerpatrol.LayerPatrol
 	tracer    Tracer
 }
 
+func (h *Hare) Register(signer *signing.EdSigner) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	h.log.Info("register signing key", zap.Stringer("node", signer.NodeID()))
+	h.signers[string(signer.PublicKey().Bytes())] = signer
```

Comment:
NIT: I think `NodeID` is more expressive here.

```suggestion
	h.signers[string(signer.NodeID().Bytes())] = signer
```